rclcpp/include/rclcpp/node.hpp

```diff
@@ -447,9 +452,16 @@ class Node : public std::enable_shared_from_this<Node>
   rclcpp::node_interfaces::NodeParametersInterface::SharedPtr
   get_node_parameters_interface();
 
+  RCLCPP_PUBLIC
+  Node::SharedPtr
+  create_sub_node(const std::string & extended_namespace_);
```

Comment:
"subnode" is nice and succinct. And we can refer to the namespace as the "sub namespace", rather than "extended". 

I'm also okay with another option. In any case, I agree that the naming should be consistent.

Reply:
I changed all the `extended_namespace` to `sub_namespace`.

On how to call the object returned by this function, I am receptive to any option that seems appropriate.